lms/envs/common.py

```diff
@@ -1603,7 +1603,20 @@
 # and defines some "args" function that returns a list with the command arguments to execute.
 REQUIRE_ENVIRONMENT = "node"
 
-
+# In production, the Django pipeline appends a file hash to JavaScript file names.
+# This makes it difficult for RequireJS to load its requirements, since module names
+# specified in JavaScript code do not include the hash.
+# For this reason, we calculate the actual path including the hash on the server
+# when rendering the page.  We then override the default paths provided to RequireJS
+# so it can resolve the module name to the correct URL.
+#
+# If you want to load JavaScript dependencies using RequireJS
+# but you don't want to include those dependencies in the JS bundle for the page,
+# then you need to add the module and URL path to this dictionary.
+REQUIRE_JS_PATH_OVERRIDES = {
+    'lmsCoursewareBookmarkButton': 'js/bookmarks/views/bookmark_button.js',
```

Comment:
My recommendation would be to use the path as the name too, because it will be less confusing. It is very difficult to determine where ``lmsCoursewareBookmarkButton`` is defined when looking at code that uses it. You could then just make this variable be an array of paths.